tests/phpunit/Fixtures/Console/FakeOutput.php

```diff
@@ -4,69 +4,12 @@
 
 namespace Infection\Tests\Fixtures\Console;
 
-use Infection\Tests\UnsupportedMethod;
-use Symfony\Component\Console\Formatter\OutputFormatterInterface;
-use Symfony\Component\Console\Output\OutputInterface;
-
-final class FakeOutput implements OutputInterface
-{
-    public function write($messages, $newline = false, $options = 0)
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function writeln($messages, $options = 0)
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function setVerbosity($level)
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function getVerbosity(): int
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function isQuiet(): bool
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function isVerbose(): bool
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function isVeryVerbose(): bool
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function isDebug(): bool
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function setDecorated($decorated)
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function isDecorated(): bool
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function setFormatter(OutputFormatterInterface $formatter)
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
-
-    public function getFormatter(): OutputFormatterInterface
-    {
-        throw UnsupportedMethod::method(__CLASS__, __FUNCTION__);
-    }
+use function class_alias;
+use Composer\InstalledVersions;
+use function version_compare;
+
+if (version_compare((string)InstalledVersions::getPrettyVersion('symfony/console'), 'v6.0', '<')) {
+    class_alias(FakeOutputSymfony5::class, \Infection\Tests\Fixtures\Console\FakeOutput::class);
+} else {
+    class_alias(FakeOutputSymfony6::class, \Infection\Tests\Fixtures\Console\FakeOutput::class);
```

Comment:
not sure I think my comment got erased with the forced push? Anyway I don't quite understand why this Sf6 stub is needed _now_ when it is about Symfony6? Same with the change for the Sf5 stub

Reply:
Otherwise there will be methods compatibility exception as in Symfony 7 native return types have been added https://github.com/symfony/symfony/blob/v7.0.0-BETA2/src/Symfony/Component/Console/Output/OutputInterface.php#L40

Actually this fix is needed for Symfony 7 but I realized that starting from Symfony 6 all argument types are defined in the interface either. See https://github.com/symfony/symfony/blob/6.0/src/Symfony/Component/Console/Output/OutputInterface.php#L39

But PHP doesn't complaint about this type of inconsistency and that's why we didn't get errors with Symfony 6. I basically decided that this will be okay if we add stub starting from Symfony 6.